gc3apps/ivs.uzh/gtree.py

```diff
@@ -0,0 +1,211 @@
+
+...
+
+Options:
+"""
+
+__version__ = 'development version (SVN $Revision$)'
+# summary of user-visible changes
+__changelog__ = """
+  2016-05-11:
+  * Initial version
+"""
+__author__ = 'Adrian Etter <adrian.etter@uzh.ch>'
+__docformat__ = 'reStructuredText'
+
+
+# run script, but allow GC3Pie persistence module to access classes defined here;
+# for details, see: https://github.com/uzh/gc3pie/issues/95
+if __name__ == "__main__":
+    import gtree
+    gtree.GtreeScript().run()
+
+from argparse import ArgumentError
+import os
+from os.path import basename, exists, join, realpath
+import sys
+import time
+import tempfile
+
+import shutil
+# import csv
+
+from pkg_resources import Requirement, resource_filename
+
+import gc3libs
+import gc3libs.exceptions
+from gc3libs import Application, Run, Task
+from gc3libs.cmdline import SessionBasedScript, executable_file
+import gc3libs.utils
+from gc3libs.quantity import Memory, kB, MB, MiB, GB, Duration, hours, minutes, seconds
+from gc3libs.workflow import RetryableTask
+
+DEFAULT_CORES = 4
+DEFAULT_MEMORY = Memory(7,GB)
+DEFAULT_WALLTIME = Duration(200,hours)
+
+## custom application class
+class GtreeApplication(Application):
+    """
+    Custom class to wrap the execution of a R script
+    A convenient wrapper script is provided to facilitate how the command file
+    is executed on the remote end.
+    """
+    application_name = 'gtree'
+
+    def __init__(self, input_function, input_dataSet, **extra_args):
+
+        self.output_dir = extra_args['output_dir']
+        self.input_function = input_function
+        #
+        inputs = dict()
+        # outputs = dict()
+        #
+        # gtree_wrapper_sh = resource_filename(Requirement.parse("gc3pie"),
+        #                                   "gc3libs/etc/gtree_wrapper.sh")
+        gtree_wrapper_sh = resource_filename(Requirement.parse("gc3pie"),
+                                              "gc3libs/etc/gtree_wrapper.sh")
+        #
+        inputs[gtree_wrapper_sh] = os.path.basename(gtree_wrapper_sh)
+        # inputs['foo'] = 'foo'
+        #
+        inputs[input_function] = "%s" % os.path.basename(input_function)
+        inputs[input_dataSet] = "%s" % os.path.basename(input_dataSet)
+
+        #
+        arguments = "./%s %s %s" % (inputs[gtree_wrapper_sh], inputs[input_function], inputs[input_dataSet])
+        # print 'hello'
+        Application.__init__(
+            self,
+            arguments = arguments,
+            inputs = inputs,
+            outputs = ["./results"],
+            stdout = 'gtree.log',
+            join=True,
+            executables = "./%s" % os.path.basename(input_function),
+            **extra_args)
+
+    def terminated(self):
+        """
+        Move results into original results folder
+        """
+        gc3libs.log.info("Application terminated with exit code %s" % self.execution.exitcode)
+        for result in os.listdir(os.path.join(self.output_dir,"results/")):
```

Comment:
`os.listdir()` will also return directories and special files like `.` and `..` -- take care!